source/MatchmakingGraphRoot.h

```diff
@@ -0,0 +1,144 @@
+﻿#ifndef __MATCHMAKING_GRAPH_ROOT_H__
+#define __MATCHMAKING_GRAPH_ROOT_H__
+#include <cugl/cugl.h>
+
+#include <vector>
+
+#include "BreachModel.h"
+#include "DonutModel.h"
+#include "DonutNode.h"
+#include "InputController.h"
+
+class MatchmakingGraphRoot : public cugl::Scene {
+   protected:
+	/** The asset manager for this game mode. */
+	std::shared_ptr<cugl::AssetManager> assets;
+	/** The Screen's Height. */
+	float screenHeight;
+
+	// VIEW
+	/** Button to create host */
+	std::shared_ptr<cugl::Button> host;
+	/** Button to create client */
+	std::shared_ptr<cugl::Button> client;
+	/** Textfield for entering room ID */
+	std::shared_ptr<cugl::TextField> roomInput;
+	/** Button to Enter Input */
+	std::shared_ptr<cugl::Button> textInput;
+	/** Label for room ID */
+	std::shared_ptr<cugl::Label> roomLabel;
+
+	// MODEL
+	int playerId;
+	/** RoomId for host display */
+	std::string roomId;
+
+	/**
+	 * Returns an informative string for the position
+	 *
+	 * This function is for writing the current donut position to the HUD.
+	 *
+	 * @param coords The current donut coordinates
+	 *
+	 * @return an informative string for the position
+	 */
+	std::string positionText();
+
+   public:
+#pragma mark -
+#pragma mark Constructors
+	/**
+	 * Creates a new game mode with the default values.
+	 *
+	 * This constructor does not allocate any objects or start the game.
+	 * This allows us to use the object without a heap pointer.
+	 */
+	MatchmakingGraphRoot() : Scene(), roomId("-1"), playerId(-1) {}
```

Comment:
There's got to be a better default value for a string than just `"-1"`. Perhaps the empty string instead?

Reply:
You know, for some reason that didn’t occur to me when I created it. Yeah I’ll change that.